File: ios/HMP/HMP/Features/Common/Design/TypographyTokens.swift
import SwiftUI

/// 字体令牌 - 与 Android TypographyTokens.kt 保持对等
/// sp -> pt 1:1 映射，字体族使用系统 SF Pro（HarmonyOS Sans 在 iOS 无对应，使用系统字体）
struct TypographyTokens {
    // MARK: - Display

    static let displayLarge = Font.system(size: 40, weight: .bold, design: .default)
    static let displayMedium = Font.system(size: 24, weight: .bold, design: .default)
    static let displaySmall = Font.system(size: 20, weight: .bold, design: .default)

    // MARK: - Headline

    static let headlineLarge = Font.system(size: 24, weight: .bold, design: .default)
    static let headlineMedium = Font.system(size: 18, weight: .bold, design: .default)
    static let headlineSmall = Font.system(size: 14, weight: .bold, design: .default)

    // MARK: - Title

    static let titleLarge = Font.system(size: 18, weight: .bold, design: .default)
    static let titleMedium = Font.system(size: 16, weight: .medium, design: .default)
    static let titleSmall = Font.system(size: 14, weight: .medium, design: .default)

    // MARK: - Body

    static let bodyLarge = Font.system(size: 16, weight: .regular, design: .default)
    static let bodyMedium = Font.system(size: 14, weight: .regular, design: .default)
    static let bodySmall = Font.system(size: 12, weight: .regular, design: .default)

    // MARK: - Label

    static let labelLarge = Font.system(size: 14, weight: .medium, design: .default)
    static let labelMedium = Font.system(size: 12, weight: .medium, design: .default)
    static let labelSmall = Font.system(size: 11, weight: .medium, design: .default)

    // MARK: - 行高间距常量 (pt)

    static let displayLargeLineHeight: CGFloat = 40
    static let displayMediumLineHeight: CGFloat = 32
    static let displaySmallLineHeight: CGFloat = 28
    static let headlineLargeLineHeight: CGFloat = 32
    static let headlineMediumLineHeight: CGFloat = 28
    static let headlineSmallLineHeight: CGFloat = 24
    static let titleLargeLineHeight: CGFloat = 28
    static let titleMediumLineHeight: CGFloat = 24
    static let titleSmallLineHeight: CGFloat = 20
    static let bodyLargeLineHeight: CGFloat = 24
    static let bodyMediumLineHeight: CGFloat = 20
    static let bodySmallLineHeight: CGFloat = 16
    static let labelLargeLineHeight: CGFloat = 20
    static let labelMediumLineHeight: CGFloat = 16
    static let labelSmallLineHeight: CGFloat = 16
}

File: desktop/feature-ui/src/desktopMain/kotlin/com/hmp/desktop/ui/common/design/typography/TypographyTokens.kt
package com.hmp.desktop.ui.common.design.typography

import androidx.compose.material3.Typography
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.sp

object TypographyTokens {
    // Desktop: 使用系统默认字体族 (HarmonyOS 字体在桌面端通过系统提供)
    val MyFontFamily: FontFamily = FontFamily.Default

    // 定义自定义 Typography
    val Typography = Typography(
        displayLarge = TextStyle(
            fontFamily = MyFontFamily,
            fontWeight = FontWeight.Bold,
            fontSize = 40.sp,
            lineHeight = 40.sp,
            letterSpacing = 0.25.sp
        ),
        displayMedium = TextStyle(
            fontFamily = MyFontFamily,
            fontWeight = FontWeight.Bold,
            fontSize = 24.sp,
            lineHeight = 32.sp,
            letterSpacing = 0.25.sp
        ),
        displaySmall = TextStyle(
            fontFamily = MyFontFamily,
            fontWeight = FontWeight.Bold,
            fontSize = 20.sp,
            lineHeight = 28.sp,
            letterSpacing = 0.25.sp
        ),
        headlineLarge = TextStyle(
            fontFamily = MyFontFamily,
            fontWeight = FontWeight.Bold,
            fontSize = 24.sp,
            lineHeight = 32.sp,
            letterSpacing = 0.sp
        ),
        headlineMedium = TextStyle(
            fontFamily = MyFontFamily,
            fontWeight = FontWeight.Bold,
            fontSize = 18.sp,
            lineHeight = 28.sp,
            letterSpacing = 0.sp
        ),
        headlineSmall = TextStyle(
            fontFamily = MyFontFamily,
            fontWeight = FontWeight.Bold,
            fontSize = 14.sp,
            lineHeight = 24.sp,
            letterSpacing = 0.sp
        ),
        titleLarge = TextStyle(
            fontFamily = MyFontFamily,
            fontWeight = FontWeight.Bold,
            fontSize = 18.sp,
            lineHeight = 28.sp,
            letterSpacing = 0.sp
        ),
        titleMedium = TextStyle(
            fontFamily = MyFontFamily,
            fontWeight = FontWeight.Medium,
            fontSize = 16.sp,
            lineHeight = 24.sp,
            letterSpacing = 0.25.sp
        ),
        titleSmall = TextStyle(
            fontFamily = MyFontFamily,
            fontWeight = FontWeight.Medium,
            fontSize = 14.sp,
            lineHeight = 20.sp,
            letterSpacing = 0.25.sp
        ),
        bodyLarge = TextStyle(
            fontFamily = MyFontFamily,
            fontWeight = FontWeight.Normal,
            fontSize = 16.sp,
            lineHeight = 24.sp,
            letterSpacing = 0.5.sp
        ),
        bodyMedium = TextStyle(
            fontFamily = MyFontFamily,
            fontWeight = FontWeight.Normal,
            fontSize = 14.sp,
            lineHeight = 20.sp,
            letterSpacing = 0.25.sp
        ),
        bodySmall = TextStyle(
            fontFamily = MyFontFamily,
            fontWeight = FontWeight.Normal,
            fontSize = 12.sp,
            lineHeight = 16.sp,
            letterSpacing = 0.4.sp
        ),
        labelLarge = TextStyle(
            fontFamily = MyFontFamily,
            fontWeight = FontWeight.Medium,
            fontSize = 14.sp,
            lineHeight = 20.sp,
            letterSpacing = 0.1.sp
        ),
        labelMedium = TextStyle(
            fontFamily = MyFontFamily,
            fontWeight = FontWeight.Medium,
            fontSize = 12.sp,
            lineHeight = 16.sp,
            letterSpacing = 0.5.sp
        ),
        labelSmall = TextStyle(
            fontFamily = MyFontFamily,
            fontWeight = FontWeight.Medium,
            fontSize = 11.sp,
            lineHeight = 16.sp,
            letterSpacing = 0.5.sp
        )
    )
}
